lib/jekyll-feed/generator.rb

```diff
@@ -69,6 +70,31 @@ def collections
       @collections
     end
 
+    def generate_feed_by_tag
+      tags_config = config["tags"].is_a?(Hash) ? config["tags"] : {}
+
+      excludes  = tags_config["excludes"] || []
+      includes  = tags_config["includes"] || @site.tags.keys
+      tags_pool = includes - excludes
+      tags_path = tags_config["path"] || "/feed/by_tag/"
```

Comment:
What happens if I pass in a malicious path here via the site's config? (e.g., `../../../../../some-secret-file`)?

* I _think_ `file_exists?` below could reveal the file's existence?
* It [looks like](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/page.rb#L46) we may be sanitizing the path to be _within the plugin source_? Is that possible or am I misreading?
* If that's the case, then a malicious user may be able to write to an arbitrary path (potentially within the plugin source?)

Reply:
It gets passed to the same "make_page" function as generating the main feed uses. Honestly I'm not sure what happens with malicious paths get used, I assumed that function already included sanity checking for the path. The only change I made to that function was to allow it to include the option to pass in the tag you want a feed for.

Reply:
@benbalter In reality the `@base` and `@path` values never get used:
```ruby
# jekyll/lib/jekyll/page.rb

def path
  data.fetch("path") { relative_path }
end

# The path to the page source file, relative to the site source
def relative_path
  @relative_path ||= File.join(*[@dir, @name].map(&:to_s).reject(&:empty?)).sub(%r!\A\/!, "")
end
```
`@base` which equals to the value of `__dir__` is not used in our use-case because the `:read_yaml` is overridden:
https://github.com/jekyll/jekyll-feed/blob/9cda8c4e4940cc6947f1525b23461d06b1caaa1c/lib/jekyll-feed/page-without-a-file.rb#L5-L7

That said, your observation cannot be left disregarded (also present in the `master` and published versions) since the instance variable could still be read via another third-party plugin when used outside the GitHub Pages environment.

Reply:
> What happens if I pass in a malicious path here via the site's config? (e.g., ../../../../../some-secret-file)?

Added a test for this in https://github.com/jekyll/jekyll-feed/pull/264/commits/93c69a83a3fc2a8bd80d95f5348cdb5c8eecdd7a